### PR TITLE
[DOCS] Add 7.4 breaking changes for transforms and data frame analytics

### PR DESCRIPTION
**Pull request by droberts195**
The PRs that made these changes are:

- https://github.com/elastic/elasticsearch/pull/44350
- https://github.com/elastic/elasticsearch/pull/45276
- https://github.com/elastic/elasticsearch/pull/45856

Preview: http://elasticsearch_46821.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.x/breaking-changes-7.4.html

### CODE REVIEW

File: docs/reference/migration/migrate_7_4.asciidoc [hunk 1]
```diff
@@ -133,3 +133,65 @@ deprecated, and will be removed in version 8.0.0. Instead, use `node.pidfile`.
...
+
+[discrete]
+[[breaking_74_transform_changes]]
+=== {transform-cap} changes
+
+[discrete]
+[[transform_stats_format]]
+==== Stats response format changes
+
+The response format of the <<get-data-frame-transform-stats>> is very different
```
Comment (szabosteve) [inline docs/reference/migration/migrate_7_4.asciidoc]: The `data frame transform` URLs have been updated to `transform` URLs. Related PR: #46940 

```suggestion
The response format of the <<get-transform-stats>> is very different
```

File: docs/reference/migration/migrate_7_4.asciidoc [hunk 2]
```diff
@@ -133,3 +133,65 @@ deprecated, and will be removed in version 8.0.0. Instead, use `node.pidfile`.
...
+- `task_state` and `indexer_state` are combined into a single `state` field
+  that replaces the old `state` object.
+- Within the `checkpointing` object, `current` is renamed to `last` and
+  `in_progress` to `next`.
+- The `checkpoint` number is now nested under `last` and `next`.
+- `checkpoint_progress` is now reported in an object nested in the `next`
+  checkpoint object. (If there is no `next` checkpoint then no checkpoint is
+  in progress and by definition the `last` checkpoint is 100% complete.)
+
+For an example of the new format see <<get-data-frame-transform-stats-example>>.
```
Comment (szabosteve) [inline docs/reference/migration/migrate_7_4.asciidoc]: Same as above.

```suggestion
For an example of the new format see <<get-transform-stats-example>>.
```

**Review by szabosteve — COMMENTED** [review]
The `data frame transform` URLs have been updated to `transform` recently. I've added a couple of suggestions to change the URLs in the text.

After fixing the URLs, LGTM.

**Review by lcawl — APPROVED** [review]
One minor comment. Otherwise, LGTM

**Pull request merged**